proposals/nnnn-remote_atf_testing.md

```diff
@@ -0,0 +1,83 @@
+# Remote Automated testing
+
+* Proposal: [SDL-NNNN](NNNN-remote_atf_testing.md)
+* Author: [Alexander Kutsan](https://github.com/LuxoftAKutsan)
+* Status: **Awaiting review**
+* Impacted Platforms: [ATF]
+
+## Introduction
+
+This proposal is about modification of Automated Test Framework (ATF) for providing ability to test SDL on remote workstation usinig automated test scripts.   
+
+## Motivation
+
+Automated testing of SDL on the real OEM hardware would significantly reduce the amount of efforts at integration stage. 
+Also, it can ensure that during integration the business logic SDL will be ported successfully and will not be corrupted. 
+
+On various OEM platforms communication between HMI and SDL is performed through different transport, that may not be retrieved from the target (i.e. mqueue). So for ATF it's not possible to connect to SDL from HMI side and emulate HMI behavior.  
+This proposal suggests adding the optional proxy (relay) layer. This layer will connect to SDL, which is located on the OEM's target and expose API to ATF, which is located on other workstation via Transmission Control Protocol (TCP).
+
+## Proposed solution
+
+Create Remote ATF Adapter as separate application. This application will be executed on the same hardware with SDL and will provide API for communication with SDL and will control SDL life cycle.
+
+### Remote ATF Adapter
+
+ATF will use Remote ATF Adapter as a relay for sending data to SDL via HMI connection. 
+Remote ATF Adapter may utilise any local transport that is used for connection between SDL and HMI on specific OEM hardware, but it will provide TCP for ATF and Open Source ATF will be used. 
+
+![Remote ATF Adapter](/assets/proposals/nnnn-hmi-relay/ATFRemoteAdapter.png)
+
+Remote ATF Adapter should provide the following functionality:
+
+#### Connection management:
+For sending and receiving data. In some cases more than one connection (with different connection types) should be opened, so interface should be rather general.
+
+The following RPCs are required: 
+ - Open Connection (connection parameters).
+ - send (connection, data) -> status.
+ - receive (connection) -> data. 
+ - Close Connection(connection) -> status.
+
+### RPCs for SDL life cycle management:
+Testing of some SDL functionality (i.e. Resumption flow) requires performing of ignition cycles. For SDL ignition cycle means that it will be stopped and started again (with additional notifications before stop). So ATF need to control SDL life cycle. 
```

Comment:
Change to: 
Testing of some SDL functionality (i.e. Resumption flow) requires performing of ignition cycles. For the SDL ignition cycle, that means it will be stopped and started again (with additional notifications before stop). So ATF will need to control the SDL life cycle.